assets/js/app/Data.js

```diff
@@ -47,9 +99,19 @@ LocusZoom.Data.Requester = function(sources) {
     };
 };
 
-LocusZoom.Data.AssociationSource = function(url) {
-    this.url = url;
-
+LocusZoom.Data.AssociationSource = function(init) {
+    if (LocusZoom.isString(init)) {
+        this.url = init;
+        this.params = {};
+    } else {
+        this.url = init.url;
+        this.params = init.params || {};
+    }
+    if (!this.url) {
+        throw("AssociationSource not initialized with required URL");
+    }
+    
+    var that=this;
```

Comment:
Other libraries shouldn't be affected by this because each time you bind variables to a function's scope you do it as you define that function and it only applies to the scope of that function. In other wordsm binding `this` on a function you define that invokes methods on the Q library will have no effect on when those Q methods use the `this` keyword because your binding only applies to the scope of the function you created. So this should play nice with everything. =)